packages/react-components/react-portal/src/components/Portal/usePortal.ts

```diff
@@ -26,6 +26,9 @@ export const usePortal_unstable = (props: PortalProps): PortalState => {
     if (state.virtualParentRootRef.current && state.mountNode) {
       setVirtualParent(state.mountNode, state.virtualParentRootRef.current);
     }
+    return () => {
+      if (state.mountNode) {cleanVirtualParent(state.mountNode);}
```

Comment:
```suggestion
      if (state.mountNode) {setVirtualParent(undefined);}
```
This should also work

Reply:
+1 let's take this approach to save on bundle size

Reply:
there's also no need for this to be conditional either AFAIK, you could just clear the previous virtual parent after every effect run

Reply:
@ling1726 Typescript complains if I do not check first. The types don't match.  `state.mountNode` can be null and `setVirtualParent` does not accept a possible null value as first parameter